### PR TITLE
feat: Add IndicatorPlacement to TabBar

### PR DESCRIPTION
**Pull request by kazo0**
related https://github.com/unoplatform/uno.chefs/issues/382

## PR Type

What kind of change does this PR introduce?
- Feature

Adds ability to drive whether or not the selection indicator for a `TabBar` will display above over below the content of the `TabBar`

With `IndicatorPlacement.Above`:

image

With `IndicatorPlacement.Below`:

image

### CODE REVIEW

File: doc/controls/TabBarAndTabBarItem.md [hunk 1]
```diff
@@ -108,6 +108,7 @@ SelectionIndicatorContent|object|Gets or sets the content to be displayed as the
 SelectionIndicatorContentTemplate|DataTemplate|Gets or sets the data template that is used to display the content of the selection indicator
 SelectionIndicatorPresenterStyle|Style|Gets or sets the style to be applied for the content of the `TabBarSelectionIndicatorPresenter`
 SelectionIndicatorTransitionMode|IndicatorTransitionMode|Gets or sets the behavior of the selection indicator. The indicator can either slide or snap to the newly selected item. Defaults to `Snap`
+SelectionIndicatorPlacement|IndicatorPlacement|Gets or sets the placement of the selection indicator. The indicator can either be display `Above` or `Below` the TabBarItem Content. Defaults to `Above`
```
Comment (agneszitte-nventive) [inline doc/controls/TabBarAndTabBarItem.md:111]:
```suggestion
SelectionIndicatorPlacement|IndicatorPlacement|Gets or sets the placement of the selection indicator. The indicator can either be displayed `Above` or `Below` the TabBarItem Content. Defaults to `Above`
```

File: src/Uno.Toolkit.RuntimeTests/Tests/TabBarTests.cs [hunk 2]
```diff
@@ -261,5 +265,185 @@ public async Task Verify_Indicator_Transitions(Orientation orientation, Indicato
...
+				await UnitTestsUIContentHelper.WaitForIdle();
+
+				var selectedItem = SUT.ContainerFromItem(SUT.SelectedItem) as TabBarItem;
+				Assert.IsNotNull(selectedItem);
+
+				var renderer = await TakeScreenshot(SUT);
+				var centerPoint = selectedItem!.TransformToVisual(SUT).TransformPoint(new Point(selectedItem.ActualWidth / 2, selectedItem.ActualHeight / 2));
+				var pixel = await GetColorAt(renderer, (int)centerPoint.X, (int)centerPoint.Y);
+ 
+				AssertExpectedColor(Colors.Red, pixel);
```
[the remark this answers is not in the document]
Comment (Xiaoy312) [reply src/Uno.Toolkit.RuntimeTests/Tests/TabBarTests.cs:310]: Also please include a message with the assertion, so we get a more comprehensive test result.

File: src/Uno.Toolkit.RuntimeTests/Tests/TabBarTests.cs [hunk 3]
```diff
@@ -261,5 +265,185 @@ public async Task Verify_Indicator_Transitions(Orientation orientation, Indicato
...
+					{
+						Padding = new Thickness(20),
+						Background = new SolidColorBrush(Colors.Green),
+					}
+				};
+			}
+		}
+
+		[TestMethod]
+		[DataRow(true, DisplayName = "Verify Indicator Above")]
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.RuntimeTests/Tests/TabBarTests.cs:334]: you can pass also use the IndicatorPlacement enum here

File: src/Uno.Toolkit.RuntimeTests/Tests/TabBarTests.cs [hunk 4]
```diff
@@ -261,5 +265,185 @@ public async Task Verify_Indicator_Transitions(Orientation orientation, Indicato
...
+			}
+		}
+
+		private static void AssertExpectedColor(Color expected, Color? actual)
+		{
+			Assert.AreEqual(expected.A, actual?.A);
+			Assert.AreEqual(expected.R, actual?.R);
+			Assert.AreEqual(expected.G, actual?.G);
+			Assert.AreEqual(expected.B, actual?.B);
+		}
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.RuntimeTests/Tests/TabBarTests.cs:402]: this and the helper methods below, moving them into a helper separate `RuntimeImageAssert` class

File: src/Uno.Toolkit.RuntimeTests/Tests/TabBarTests.cs [hunk 5]
```diff
@@ -261,5 +265,185 @@ public async Task Verify_Indicator_Transitions(Orientation orientation, Indicato
...
+				await renderer.RenderAsync(element);
+				return renderer;
+			}
+			else
+			{
+				return null;
+			}
+		}
+
+		private static bool CanTakeScreenshot()
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.RuntimeTests/Tests/TabBarTests.cs:438]:
```suggestion
		private static bool IsScreenshotSupport()
```
or just `RuntimeImageAssert::IsSupported()`

File: src/Uno.Toolkit.UI/Controls/TabBar/TabBar.xaml [hunk 6]
```diff
@@ -75,28 +75,58 @@
...
-										<Setter Target="SelectionIndicatorPresenter.HorizontalContentAlignment" Value="Stretch" />
+										<Setter Target="BelowSelectionIndicatorPresenter.VerticalContentAlignment" Value="Top" />
+										<Setter Target="BelowSelectionIndicatorPresenter.HorizontalContentAlignment" Value="Stretch" />
+										<Setter Target="AboveSelectionIndicatorPresenter.VerticalContentAlignment" Value="Top" />
+										<Setter Target="AboveSelectionIndicatorPresenter.HorizontalContentAlignment" Value="Stretch" />
+									</VisualState.Setters>
+								</VisualState>
+							</VisualStateGroup>
+							<VisualStateGroup x:Name="IndicatorPlacementStates">
+								<VisualState x:Name="Above">
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.UI/Controls/TabBar/TabBar.xaml:94]: given Above is the dp default value, you can have these setters as the default states,
and leave this empty

File: src/Uno.Toolkit.RuntimeTests/Tests/TabBarTests.cs
Comment (Xiaoy312) [inline src/Uno.Toolkit.RuntimeTests/Tests/TabBarTests.cs:1]: overall, the helper class should really be in runtime test engine (unoplatform/uno.ui.runtimetests.engine#87)
so we could let this code as is for now, just tag a fixme/todo in comment

**Review by Xiaoy312 — APPROVED** [review]

File: src/Uno.Toolkit.RuntimeTests/Tests/TabBarTests.cs
Comment (kazo0) [reply src/Uno.Toolkit.RuntimeTests/Tests/TabBarTests.cs:1]: Yeah that's what I figured since it _should_ be temporary

**Pull request merged**